test/boost/memtable_test.cc

```diff
@@ -644,6 +638,74 @@ SEASTAR_THREAD_TEST_CASE(test_tombstone_merging_with_multiple_versions) {
         .produces(m1 + m2 + m3);
 }
 
+SEASTAR_THREAD_TEST_CASE(test_tombstone_merging_with_mvcc_and_preemption) {
+    tests::reader_concurrency_semaphore_wrapper semaphore;
+    simple_schema ss;
+    auto s = ss.schema();
+    auto mt = make_lw_shared<replica::memtable>(ss.schema());
+
+    auto pk = ss.make_pkey(0);
+    auto pr = dht::partition_range::make_singular(pk);
+
+    // Produce large m0 so that merging range tombstone from m1 into m0 is likely to be preempted in the middle.
```

Comment:
I chose the size intuitively, and verified that it reproduces the bug which this test guards against, before I fixed the bug.

Agree that it's not a very reliable way to test it, still better than none.

We could check if preemption happens, but since the test is not reliable, that could introduce flakyness.